YT-Publish WOH: Fail more aggressively

Arnei opened the pull request:
Currently the Youtube Publish WOH is a bit meek, often only logging little warnings when it fails to publish. This makes the WOH more assertive by having it throw exceptions instead, properly failing the workflow.

Arnei committed:
YT-Publish WOH: Fail more aggressively

Currently the Youtube Publish WOH is a bit meek, often only logging
little warnings when it fails to publish. This makes the WOH more
assertive by having it throw exceptions instead, properly failing the
workflow.

JulianKniephoff reviewed (approved):
Code lgtm. Will try to get a test in later as well.

JulianKniephoff commented on modules/distribution-workflowoperation/src/main/java/org/opencastproject/workflow/handler/distribution/PublishYouTubeWorkflowOperationHandler.java:
```diff
@@ -122,13 +122,12 @@ public WorkflowOperationResult start(final WorkflowInstance workflowInstance, Jo
       if (elements.size() < 1) {
-        logger.info("No mediapackage element was found for publishing");
-        return createResult(mediaPackage, Action.CONTINUE);
+        throw new WorkflowOperationException("No mediapackage element was found for publishing");
```
I tested at least these and they indeed work and fail the workflow. 👌

KatrinIhler commented on modules/distribution-workflowoperation/src/main/java/org/opencastproject/workflow/handler/distribution/PublishYouTubeWorkflowOperationHandler.java:
```diff
@@ -122,13 +122,12 @@ public WorkflowOperationResult start(final WorkflowInstance workflowInstance, Jo
       if (elements.size() < 1) {
-        logger.info("No mediapackage element was found for publishing");
-        return createResult(mediaPackage, Action.CONTINUE);
+        throw new WorkflowOperationException("No mediapackage element was found for publishing");
```
Should this really fail? I think "Skipped" was the right result in this case. We mark other operations as skipped when the selector doesn't match any elements as well.

Arnei replied:
Fair enough, might've been a little too eager there.

KatrinIhler replied:
Actually I just realized that it returns Action.CONTINUE there. I think it should be Action.SKIP to make it obvious nothing happened. Could you do that? :sweat_smile: I can't make a suggestion on unchanged code unfortunately.

Arnei replied:
Can do that. Not the scope of this PR, but should it be SKIP in other WOHs too then? Like https://github.com/opencast/opencast/blob/cfc7bf1b6d7b713d71e107041844338075c0b38a/modules/distribution-workflowoperation/src/main/java/org/opencastproject/workflow/handler/distribution/PublishEngageWorkflowOperationHandler.java#L379

KatrinIhler replied:
In my opinion yes. I already did this for the tagging operation a while ago, see #5108. But unfortunately this is wildly inconsistent right now.

KatrinIhler reviewed (changes requested):
I don't agree with one of the cases (see ~~below~~ above), the rest looks fine.

Arnei committed:
Don't throw exception in YT-Pub-WOH if no MPEs for publishing

It is actually perfectly fine for a workflow to result in no
elements for publishing, so we should not throw an exception over it.

Arnei committed:
Change `CONTINUE` to `SKIP` in YT Publish WOH

Since we didn't really do anything, skip seems more appropriate.

KatrinIhler reviewed (approved):
Great!
